File: src/main/java/com/ferdingler/Bar.java
package com.ferdingler;

import java.util.Objects;

public class Bar {
 
    public static final String BEER = "Corona";
    public static final String JUICE = "Orange Juice";
    public static final String COKE = "Coca Cola";
    public static final String WATER = "Topo Chico";
    public static final String MOJITO = "Rum Mojito";
 
    public int processOrder(String drink, int quantity, boolean isStudent) {
 
        if (quantity > 2 && (Objects.equals(drink, BEER) || Objects.equals(drink, MOJITO))) {
            throw new RuntimeException("Too much alcohol, max 2 per person");
        }
 
        int price;
 
        if (drink.equals(BEER)) {
            price = 70;
        } else if (drink.equals(JUICE)) {
            price = ingredient3() + 50;
        } else if (drink.equals(COKE)) {
            price = 110;
        } else if (drink.equals(MOJITO)) {
            price = ingredient1() + ingredient2() + ingredient4();
        } else if (drink.equals(WATER)) {
            price = 10;
        } else {
            throw new RuntimeException("No such drink exists");
        }
        
        if (isStudent) {
            price = price - price/10;
        }
 
        return price * quantity;
    }
 
    // one unit of rum
    private int ingredient1() {
        return 65;
    }
 
    // one unit of grenadine
    private int ingredient2() {
        return 10;
    }
 
    // one unit of orange juice
    private int ingredient3() {
        return 10;
    }
    
    //one unit of soda water
    private int ingredient4() {
        return 10;
    }
}

File: src/main/kotlin/com/ferdingler/Bar.kt
package com.ferdingler

class Bar {

    val BEER = "Corona"
    val JUICE = "Orange Juice"
    val COKE = "Coca Cola"
    val WATER = "Topo Chico"
    val MOJITO = "Rum Mojito"

    fun processOrder(drink: String, quantity: Int, isStudent: Boolean): Int {
        if (quantity > 2 && (drink == BEER || drink == MOJITO)) {
            throw RuntimeException("Too much alcohol, max 2 per person")
        }
        var price: Int
        price = if (drink == BEER) {
            70
        } else if (drink == JUICE) {
            ingredient3() + 50
        } else if (drink == COKE) {
            110
        } else if (drink == MOJITO) {
            ingredient1() + ingredient2() + ingredient4()
        } else if (drink == WATER) {
            10
        } else {
            throw RuntimeException("No such drink exists")
        }

        if (isStudent) {
            price = price - price / 10
        }

        return price * quantity
    }

    // one unit of rum
    private fun ingredient1(): Int {
        return 65
    }

    // one unit of grenadine
    private fun ingredient2(): Int {
        return 10
    }

    // one unit of orange juice
    private fun ingredient3(): Int {
        return 10
    }

    //one unit of soda water
    private fun ingredient4(): Int {
        return 10
    }

}